src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -100,6 +102,28 @@ public void Prepare(CancellationToken token)
                     info.NewRuleSetFilePath = newRuleSetFilePath;
                 }
             }
+
+            EnsureNoConflictingAdditionalFile();
+        }
+
+        private void EnsureNoConflictingAdditionalFile()
+        {
+            var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
```

Comment:
Minor inconsistency - the other services are being fetched in the constructor.

I wouldn't change this now. I'd consider changing the constructor at a later stage so it doesn't use the service provider at all - the class that constructs it has almost of required services anyway.